### PR TITLE
Pathy related changes

### PR DESCRIPTION
**Pull request by drostron**

### CODE REVIEW

File: core/src/main/scala/quasar/fs/SandboxedPathy.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+  * Use with care. Functions make the assumption that Sandboxed Pathy paths do not contain ParentIn or Current.
+  * This can not currently be guaranteed.
+  */
+object SandboxedPathy {
+
+  implicit val fileNameEqual: Equal[FileName] = Equal.equalA
+  implicit val dirNameEqual: Equal[DirName] = Equal.equalA
+
+  def rootSubPath(depth: Int, p: APath): APath = {
```
Comment (mossprescott) [inline core/src/main/scala/quasar/fs/SandboxedPathy.scala]: Can this signatur have `[T]`? Otherwise it creates work for the caller 2/3 of the time.

Comment (mossprescott) [inline core/src/main/scala/quasar/fs/SandboxedPathy.scala]: Hmm. No, I suppose not. Nevermind. This is all exposing pathy's types as somewhat of a fool's errand, I'm afraid.

File: core/src/test/scala/quasar/fs/path.scala [hunk 2]
```diff
@@ -298,4 +300,15 @@ class PathSpecs extends Specification with DisjunctionMatchers {
...
     }
   }
+
+  "Path conversion" should {
+    import pathy.Path._
+    import pathy.scalacheck.PathyArbitrary._
+
+    "preserve Pathy path when translated to Path and back" !
+    prop { aFile: AFile =>
+      refineType(Path.fromAPath(aFile).asAPath) ==== aFile.right
```
Comment (mossprescott) [inline core/src/test/scala/quasar/fs/path.scala]: *Very* happy to see this. Any reason not to have a test for `ADir` as well?

Comment (drostron) [inline core/src/test/scala/quasar/fs/path.scala]: Good point, will update to `APath`.

**Comment by jedesah** [comment]
@mossprescott This is ready for review again.

**Comment by mossprescott** [comment]
LGTM

**Pull request merged**